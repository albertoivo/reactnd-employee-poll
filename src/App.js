import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import Spinner from 'react-bootstrap/Spinner'
import Login from './components/Login'
import Private from './components/Private'
import { handleInitialData } from './actions/shared'

function App({ authedUser, loadingBar, dispatch }) {

	useEffect(() => {
		dispatch(handleInitialData())
	}, [dispatch]);

	return (loadingBar.default === undefined)
		? <div className="d-flex justify-content-center">
			<Spinner animation="border" role="status" variant="secondary" className="my-5">
				<span className="sr-only">Loading...</span>
			</Spinner>
		</div>
		: <>{!authedUser ? <Login /> : <Private />}</>
}

function mapStateToProps({ authedUser, loadingBar }) {
	return { authedUser, loadingBar }
}

export default connect(mapStateToProps)(App)
