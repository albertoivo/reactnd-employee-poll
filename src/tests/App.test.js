import React from 'react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from '../App';
import { Provider } from 'react-redux'
import rootReducer from '../reducers'
import thunk from 'redux-thunk'
import { applyMiddleware, createStore } from 'redux'

const store = createStore(rootReducer, (applyMiddleware(thunk)))

/**
 * Test with Snapshot
 */
describe('App index', () => {
    it('matches the snapshot with Redux', () => {
        var component = render(
            <Provider store={store}>
                <App />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });

});