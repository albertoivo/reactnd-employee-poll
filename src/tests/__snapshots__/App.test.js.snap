// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App index matches the snapshot with Redux 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="justify-content-center align-items-center min-vh-100 row"
      >
        <div
          class="col-md-4 col-12"
        >
          <div
            class="text-center card bg-light"
          >
            <div
              class="card-header"
            >
              User
            </div>
            <div
              class="card-body"
            >
              <form
                class=""
              >
                <div
                  class="form-group"
                >
                  
                  <select
                    class="form-control"
                    data-testid="user-select"
                    id="formGridState"
                  >
                    <option
                      value=""
                    >
                      Select user
                    </option>
                  </select>
                </div>
                <button
                  class="btn btn-outline-dark"
                  data-testid="submit-button"
                  type="submit"
                >
                  Authenticate
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="justify-content-center align-items-center min-vh-100 row"
    >
      <div
        class="col-md-4 col-12"
      >
        <div
          class="text-center card bg-light"
        >
          <div
            class="card-header"
          >
            User
          </div>
          <div
            class="card-body"
          >
            <form
              class=""
            >
              <div
                class="form-group"
              >
                
                <select
                  class="form-control"
                  data-testid="user-select"
                  id="formGridState"
                >
                  <option
                    value=""
                  >
                    Select user
                  </option>
                </select>
              </div>
              <button
                class="btn btn-outline-dark"
                data-testid="submit-button"
                type="submit"
              >
                Authenticate
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
